### ISSUE TITLE
Stable sort of server/channel/query list

### ISSUE THREAD

**Issue by djasa** [SrainApp/srain#375]
Hi, I'm using Srain 2nd day only. Now I realized I replied to different person than I thought - because #263 hid the other's user name in notification, srain flipped their position in channel list which I peripherally saw and then I somehow managed to ignore their nick at the top of their bubbles in main chat window. Being a [spatial navigator](https://humantransit.org/2009/10/global-impositioning-systems.html), constantly changing order of servers/channels/queries is ... unhelpful to me. If there was a config option to get stable sorting (optimally with groups of channels and queries of each respective server, but it doesn't matter much, main thing is that it changes only when I open or close channel/query/server), that would quite add to usability of srain for me.

In addition, grouping by server could allow removing it from respective lines and thus allow having more entries in viewport. This isn't issue for me with 10 entries but if it was 20, it would start bugging me as some of them would be out of viewport with no sensible reason (to my eyes ofc).

### version
```
$ flatpak info im.srain.Srain

Srain - Modern IRC client written in GTK

          ID: im.srain.Srain
         Ref: app/im.srain.Srain/x86_64/stable
        Arch: x86_64
      Branch: stable
     Version: 1.4.0
     License: GPL-3.0
      Origin: flathub
  Collection: org.flathub.Stable
Installation: system
   Installed: 915,5 kB
     Runtime: org.gnome.Platform/x86_64/42
         Sdk: org.gnome.Sdk/x86_64/42

      Commit: 2055941c2b9d402c1c383973df7afe8f511fd439484420746024afe9e5539f0d
      Parent: 7db333a933b6921fe1a5ebf850fc1e195f2378e9ee0e878b3a6134f2e12a3c2a
     Subject: Update to 1.4.1 (653ddbd7)
        Date: 2022-08-19 16:23:22 +0000
```

**Comment by SilverRainZ** [comment]
Hi, thanks for reporting.

> because https://github.com/SrainApp/srain/issues/263 hid the other's user name in notification

This bug should cause you to see unexpected HTML code instead of username being hidden, what notification daemon are you using? 

>  If there was a config option to get stable sorting

In fact, sorting by the newest message is a feature of Srain, but I think it is reasonable to add such a config option.

> In addition, grouping by server could allow removing it from respective lines and thus allow having more entries in viewport.

There is alreay an option for hiding server's buffer, see #361

**Comment by djasa** [comment]
Hi,

Dne čt 1. 9. 2022 8:09 uživatel Shengyu Zhang ***@***.***>
napsal:

> Hi, thanks for reporting.
>
> because #263 <https://github.com/SrainApp/srain/issues/263> hid the
> other's user name in notification
>
> This bug should cause you to see unexpected HTML code instead of username
> being hidden, what notification daemon are you using?
>
Whatever is default in gnome 42(?) in Fedora 37. I can give you better
answer when I get back to the computer 🙂

> If there was a config option to get stable sorting
>
> In fact, sorting by the newest message is a feature of Srain, but I think
> it is reasonable to add such a config option.
>
> In addition, grouping by server could allow removing it from respective
> lines and thus allow having more entries in viewport.
>
> There is alreay an option for hiding server's buffer, see #361
> <https://github.com/SrainApp/srain/issues/361>
>
I want to see server buffer and I'd actually like to see server messages
there, both to see channel conversations without unnecessary noise and in
case I want to see connection status and errors history, to have all of
them at the same tab.

As per vertical space, i meant move from current view (with server tabs
hidden)

#channel1 server
Msg preview 1
#channel2 server
Msg preview 2
#channel3 server
Msg preview 3

To:

Server1
#channel1 Msg preview 1
#channel2 Msg preview 2
#channel3 Msg preview 3

Allows to save pretty much one line per channel for servers with more than
few associated channels or queries. Vertical screen space is more scarce
resource than horizonal since advent of 16:9 screens years ago...

**Comment by SilverRainZ** [comment]
This feature has been implemented in https://github.com/SrainApp/srain/pull/421 and will be shipped in 1.8.0.
